### PR TITLE
[ISSUE #490] fix new regis logic not compitable old sdk version

### LINKED ISSUE #490: 创建子账户可以与主账户同名导致主账户登陆失败
**Describe the bug**

**To Reproduce**
Steps to reproduce the behavior.

**Expected behavior**

**Environment**
- Version: [e.g. v1.0.0]
- OS: [e.g. CentOS8]

**Additional context**

### PR DESCRIPTION
**Pull request by chuntaojun**
**Please provide issue(s) of this PR:**
Fixes #490 #457 

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration
- [ ] Docs
- [x] Installation
- [x] Performance and Scalability
- [x] Naming
- [ ] HealthCheck
- [x] Test and Release

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: apiserver/grpcserver/base.go [hunk 1]
```diff
@@ -349,10 +349,11 @@ func (b *BaseGrpcServer) AllowAccess(method string) bool {
 }
 
 // ConvertContext 将GRPC上下文转换成内部上下文
-func ConvertContext(ctx context.Context) context.Context {
+func ConvertContext(ctx context.Context, externalHeaders ...string) context.Context {
```
Comment (andrewshan) [inline apiserver/grpcserver/base.go:352]: 不要加这个参数，太奇怪了，context里面直接获取头信息就行了

File: apiserver/grpcserver/discover/client_access.go [hunk 2]
```diff
@@ -40,16 +40,18 @@ func (g *GRPCServer) ReportClient(ctx context.Context, in *api.Client) (*api.Res
 
 // RegisterInstance 注册服务实例
 func (g *GRPCServer) RegisterInstance(ctx context.Context, in *api.Instance) (*api.Response, error) {
+	externalHeaders := []string{"internal-instance-asyncregis"}
```
Comment (andrewshan) [inline apiserver/grpcserver/discover/client_access.go:43]: 不要这样写，每次初始化一个数组？

Comment (andrewshan) [inline apiserver/grpcserver/discover/client_access.go:43]: key就叫async-register，不要搞太长

File: apiserver/grpcserver/discover/client_access.go [hunk 3]
```diff
@@ -40,16 +40,18 @@ func (g *GRPCServer) ReportClient(ctx context.Context, in *api.Client) (*api.Res
...
+	rCtx := grpcserver.ConvertContext(ctx, externalHeaders...)
 	rCtx = context.WithValue(rCtx, utils.StringContext("operator"), ParseGrpcOperator(ctx))
 
 	// 客户端请求中带了 token 的，优先已请求中的为准
 	if in.GetServiceToken().GetValue() != "" {
 		rCtx = context.WithValue(rCtx, utils.ContextAuthTokenKey, in.GetServiceToken().GetValue())
 	}
 
-	if in.GetHealthCheck().GetHeartbeat().GetTtl() != nil {
+	if v, _ := rCtx.Value(utils.StringContext(externalHeaders[0])).(string); v == "true" {
```
Comment (andrewshan) [inline apiserver/grpcserver/discover/client_access.go:54]: 这不用判断是否有值，只要有带入这个header，就算异步注册

File: service/batch/config.go [hunk 4]
```diff
@@ -44,44 +44,54 @@ type CtrlConfig struct {
 	MaxBatchCount int `mapstructure:"maxBatchCount"`
 	// 写store的并发协程数
 	Concurrency int `mapstructure:"concurrency"`
+	// 抛弃过期的任务
+	DropExpireTask bool `mapstructure:"dropExpireTask"`
```
Comment (andrewshan) [inline service/batch/config.go:48]: TaskLife不填，默认就应该不用抛弃吧？所以这个标识是不是不需要？

File: service/batch/future.go [hunk 5]
```diff
@@ -77,21 +92,28 @@ func (future *InstanceFuture) Instance() *model.Instance {
...
+}
+
 // Code 获取code
 func (future *InstanceFuture) Code() uint32 {
 	return future.code
 }
 
 // sendReply 批量答复futures
 func sendReply(futures interface{}, code uint32, result error) {
+	cur := time.Now()
```
Comment (andrewshan) [inline service/batch/future.go:107]: time.Now比较耗性能，server我之前加了几个获取到毫秒粒度的高性能时间方法，可以看看

File: service/batch/config.go (hunk 4 above)
Comment (chuntaojun) [reply service/batch/config.go:48]: 也是，可以

**Review by andrewshan — APPROVED** [review]
LGTM

**Pull request merged**